storage/remote/client.go

```diff
@@ -322,22 +330,41 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryRe
 	if err != nil {
 		return nil, errors.Wrap(err, "error sending request")
 	}
+
+	if httpResp.StatusCode/100 != 2 {
+		httpResp.Body.Close()
+		return nil, errors.Errorf("remote server %s returned HTTP status %s: %s", c.url.String(), httpResp.Status, strings.TrimSpace(string(compressed)))
+	}
+
+	contentType := httpResp.Header.Get("Content-Type")
+	if strings.HasPrefix(contentType, "application/x-protobuf") {
+		return c.handleSampledResponse(httpResp, start, req, sortSeries)
+	}
+
+	if !strings.HasPrefix(contentType, "application/x-streamed-protobuf; proto=prometheus.ChunkedReadResponse") {
+		httpResp.Body.Close()
+		return nil, errors.Errorf("not supported remote read content type: %s", contentType)
+	}
+
+	// Handling streamed response
```

Comment:
Let's have comments a full sentence 🤗 
```suggestion
	// Handling streamed response.
```